cpp/src/arrow/python/arrow_to_pandas.cc

```diff
@@ -1097,6 +1081,40 @@ struct ObjectWriterVisitor {
     return Status::OK();
   }
 
+  template <typename Type>
+  enable_if_t<std::is_same<Type, MonthDayNanoIntervalType>::value, Status> Visit(
+      const Type& type) {
+    OwnedRef args(PyTuple_New(0));
+    OwnedRef kwargs(PyDict_New());
+    RETURN_IF_PYERROR();
+    auto to_date_offset = [&](const MonthDayNanoIntervalType::MonthDayNanos& interval,
+                              PyObject** out) {
+      DCHECK(internal::BorrowPandasDataOffsetType() != nullptr);
+      // TimeDelta objects do not add nanoseconds component to timestamp.
+      // so convert microseconds and remainder to preserve data
+      // but give users more expected results.
```

Comment:
Code looks ok, but I don't understand this comment. What do you mean with "do not add nanoseconds component to timestamp"?

Reply:
I'll try to clarify but there are actually two bugs:
https://github.com/pandas-dev/pandas/issues/36589
https://github.com/pandas-dev/pandas/issues/43892

This is meant to mitigate the latter.  I'll link the second issue here.

CC @tswast as an FYI

Reply:
Rephrased (had the wrong time should have been DateOffset) and referenced the bug.

Reply:
Ah, thank you.